pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -215,12 +218,13 @@ func (l *listUsersQuery) expand(
 	}
 
 	relationRewrite := relation.GetRewrite()
+	req.Depth--
```

Comment:
It seems odd we're mutating the incoming (parent) request before delegating to `l.expandRewrite(..., req, ...)`. I would expect this to lead to potentially sensitive code as far as race conditions are concerned.

I would think a safer call pattern would include something like:

```go
var req ListUsersRequest // parent request
childReq := req.Clone() // deep copy
childReq.Depth--
return l.expandRewrite(..., childReq, ...)
```

Reply:
I don't have any concerns with respect to race conditions - the request is always cloned before doing another expansion.

What I don't get is why this `req.Depth--` is down here and not immediately before the check?

```go
if req.GetDepth() == limit {
	return graph.ErrResolutionDepthExceeded
} else {
	req.Depth ++
}
```

Reply:
@jon-whit It's a good consideration but as Maria noted above, the request is cloned before the `expand` function so I'm not worried about mutation here. 

@miparnisari Changed to reflect your suggestion.